shared/src/business/useCases/caseWorksheet/updateCaseWorksheetInteractor.ts

```diff
@@ -0,0 +1,53 @@
+import {
+  CaseWorksheet,
+  RawCaseWorksheet,
+} from '@shared/business/entities/caseWorksheet/CaseWorksheet';
+import {
+  ROLE_PERMISSIONS,
+  isAuthorized,
+} from '../../../authorization/authorizationClientService';
+import { UnauthorizedError } from '@web-api/errors/errors';
+import { User } from '@shared/business/entities/User';
+
+export const updateCaseWorksheetInteractor = async (
+  applicationContext: IApplicationContext,
+  {
+    worksheet,
+  }: {
+    worksheet: RawCaseWorksheet;
+  },
+): Promise<RawCaseWorksheet> => {
+  const user = applicationContext.getCurrentUser();
+
+  if (!isAuthorized(user, ROLE_PERMISSIONS.CASE_WORKSHEET)) {
+    throw new UnauthorizedError('Unauthorized');
+  }
+
+  const rawUser = await applicationContext.getPersistenceGateway().getUserById({
+    applicationContext,
+    userId: user.userId,
+  });
+
+  const userEntity = new User(rawUser);
+
+  let judgeUser;
+  if (userEntity.isChambersUser() && !userEntity.isJudgeUser()) {
+    judgeUser = await applicationContext
+      .getUseCaseHelpers()
+      .getJudgeInSectionHelper(applicationContext, {
+        section: userEntity.section,
+      });
+  }
```

Comment:
It is repeated! Panda squad chatted about DRY-ing this up but actually felt it would obfuscate the work being done in this endpoint more than it would help anything.